backend/src/shared/configs/configuration-factory.ts

```diff
@@ -16,5 +16,7 @@ export const configurationFactory = () => {
     clerkSecretKey: process.env.CLERK_SECRET_KEY,
 
     segmentWriteKey: process.env.SEGMENT_WRITE_KEY,
+
+    openaiApiKey: process.env.OPENAI_API_KEY,
```

Comment:
We will need to add the OPENAI_API_KEY to the env.example file and add instructions in our docs for providing the new secret and enabling automatic icon, color and descriptions